### ISSUE TITLE
Disable RefManageR check.entries prior to writing to bib file

### ISSUE THREAD

**Issue by gadenbuie** [crsh/citr#42]
Related to #28, I find that RefManageR's expected fields checker is a little bit too strict for me. Currently it seems the addin sets `check.entries = FALSE`, but after attempting to write to the local bib file. 

https://github.com/crsh/citr/blob/c6143b049715c59fc6d3cda3ee24a4b44d2efd64/R/insert_citation.R#L472-L489

If `RefManageR::WriteBib()`, the addin crashes and the citation isn't written or inserted into the text. Moving this line

```r
old_BibOptions <- RefManageR::BibOptions(check.entries = FALSE)
```

above `RefManageR::WriteBib()` would allow the bib entry to be written and the citation key to be inserted.

As another option, I see that `check.entries` in `BibOptions()` can take `"warn"` (see https://www.rdocumentation.org/packages/RefManageR/versions/1.2.0/topics/BibOptions), this may be desirable to alert the user but still add the entry/citekey.

In the mean time, calling `RefManageR::BibOptions(check.entries = FALSE)` in the global environment before running the addin does resolve the issue.

**Comment by crsh** [comment]
I set `check.entries = FALSE` because this potentially emits a lot of warnings especially when used with Zotero. Also, I'm not sure the addin is meant to check the state of the bibliography entries.

The suggestion to move line 481 up seems sensible to me. Would you be willing to open a pull request?

When you say this is related to #28, would this fix it? I wasn't able to reproduce the problem.

**Comment by gadenbuie** [comment]
Choosing `FALSE` over `"warn"` makes sense to me, so I just moved line 481 up in PR #43.

I'm fairly certain that this solves #28 as well, as this was the exact structure of the error I was seeing (in my case the error was that an `@InProceedings` entry needed an `editor` field). OTOH, I didn't test that exact bib entry so I'm not completely sure. I did test inserting a few citations of arxiv papers I have in my library and all worked well.
